disnake/guild.py

```diff
@@ -3121,6 +3154,83 @@ async def change_voice_state(
         channel_id = channel.id if channel else None
         await ws.voice_state(self.id, channel_id, self_mute, self_deaf)
 
+    # Scheduled events
+
+    async def fetch_scheduled_events(
+        self, *, with_user_count: bool = None
+    ) -> List[GuildScheduledEvent]:
+        """|coro|
+
+        Request all scheduled events in this guild.
+
+        .. versionadded:: 2.3
+
+        Parameters
+        ----------
+        with_user_count: :class:`bool`
+            Whether to include the field with subscriber count.
+
+        Returns
+        -------
+        List[:class:`GuildScheduledEvent`]
+        """
+        array = await self._state.http.get_guild_scheduled_events(self.id, with_user_count)
+        return [GuildScheduledEvent(state=self._state, data=data) for data in array]
```

Comment:
I think we could use a better name than `array` here.

```suggestion
        events_data = await self._state.http.get_guild_scheduled_events(self.id, with_user_count)
        return [GuildScheduledEvent(state=self._state, data=data) for data in events_data]
```